src/main/java/edu/ucla/library/bucketeer/Metadata.java

```diff
@@ -0,0 +1,343 @@
+
+package edu.ucla.library.bucketeer;
+
+import java.io.File;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+
+import com.opencsv.bean.CsvBindByName;
+
+import info.freelibrary.util.Logger;
+import info.freelibrary.util.LoggerFactory;
+import info.freelibrary.util.StringUtils;
+
+import edu.ucla.library.bucketeer.utils.IFilePathPrefix;
+
+/**
+ * Formalized metadata recognized by Bucketeer.
+ * <p>
+ * <a href="https://californica.library.ucla.edu/importer_documentation/guide">Documented</a> in the Californica
```

Comment:
I have concerns about this integration point, with the Apps team using an easily-modified data model in Ruby, and us writing the same model into a Java model object. However, it's a good starting point... but I predict this is where future friction will occur.

Reply:
Yes, synchronization is a possible pain point. It's worth noting though that the mapping from CSV to Java Bean is just a matter of adding a field and an annotation so assuming the model changes, and we don't actually need to worry about the changed/added field in bucketeer, it may not be that painful. The toString method is the probably the most editing we'd need to do (and that can be improved to be more dynamic perhaps in the future... it's pretty hard coded now). But, yes, good observation on possible future pain points. Worth noting.

Reply:
But we are not implementing the same data model. Bucketeer only cares about the ARK, the location of the source file on the file system, and it's own output fields (the image URL and the success/failure of the job). It completely ignores all of the other metadata. This is the goal of microservices: implement your own data model within each service. The only thing we are sharing is an persistent ID and a (presumably) persistent file path.

Reply:
Yes, this is something about bucketeer that is not generalizable at the moment. Bucketeer also probably incorrectly also does some csv validation (is this a valid work? is this a valid collection?) As the requirement have firmed I'm less convinced that's necessary. It's something where there is room for improvement for sure (to make it more generalizable). It could store the csv as a text file in memory, and pull just those values it's interested in possibly in the future. Right now it persists the specific data model.